src/main/java/tictactoe_game/Main.java

```diff
@@ -14,4 +14,16 @@ public static Grid put(Element element, Grid grid, Coordinate coordinate) {
 
     return grid;
   }
+
+  public static Boolean cell_is_empty(Grid grid, Coordinate coordinate) {
```

Comment:
I think this method violates the SRP. This may point at us lacking an important abstraction /I'd call it `Cell`/. Try rewriting the method w/o passing the `Grid` and `Coordinate` explicitly to it.